testexec/README.md

```diff
@@ -0,0 +1,111 @@
+
+For hunks not present, nothing will be checked.
+(In other words, if you have a `stdout` hunk, but not a `stderr` hunk, whatever the stderr output is, it will be ignored by the test.)
+
+Note that not every testexec dataset has to contain _any_ of "output", "stdout", "stderr", or "exitcode":
+containing none of them means the commands in the sequence will all be run,
+and the exitcode is expected to be zero from each,
+and nothing about their stdout nor stderr will be checked.
+(If you want to mandate that their output is empty, you must create a blank data hunk to say so explicitly.)
+
+Input streams (a.k.a. "stdin") can also be specified:
+
+- "`input`" -- if present, will be fed to the stdin stream of the execution.
+
+Beyond this, you can also specify fixture filesystems:
+
+- "`fs/somedir/thefile.ext`" -- (and names of that general pattern: anything starting with "fs/")
+
+And last of all, sequences of causally related tests can be created:
+given a series of hunks with names like the following:
+
+```
+test-one/script
+test-one/then-test-two/script
+test-one/then-test-two/then-test-three/script
+test-one/then-test-two-b/script
+```
+
+In a scenario like the above, the nested "then-*" tests will inherit the working directory from their parent --
+or more specifically, a *copy* of it.
+So: the script for `then-test-two` receives whatever the filesystem state left behind by `test-one` was;
+the `then-test-three` script receives whatever filesystem state left behind by `then-test-two` was;
+and `then-test-two-b` receives the filesystem as it was *after `test-one`* (!  _not_ what was left by `then-test-three`).
+(This lets you do one-time setup work, or build narratives -- while also still keeping tests isolated.)
+
+
+Examples
+--------
+
+See the `selfexercise.md` file in this directory for fully worked examples.
+
+See the `testexec_test.go` file for the complete wiring to execute the examples --
+including the autopatcher support.
+
+
+Autopatching
+------------
+
+**Yes** -- this testexec implementation supports automatic fixture regeneration.
+
+There's a tiny amount of setup required:
+you do have to create a PatchAccumulator,
+assign it to the `Tester{}.Patches` field,
+and then tell the PatchAccumulator where to write any updates to.
+(This is the norm for use of the patching system.)
+(PRs welcome for further automating this.)
+Once that setup is done:
+
+Just write your 'script' hunk,
+and whichever of the 'output', 'stdout', 'stderr', or 'exitcode' hunks that you want to track,
+and run your tests like `go test ./... -testmark.regen`.
+The document will be patched with the actual output inserted into those hunks.
+
+Note that regen mode will only update hunks that already exist; it won't add them.
+(E.g., if you only have a `stdout` hunk, regen won't _add_ a `stderr` hunk.)
+
+
+Configurable Exec Hooks
+-----------------------
+
```

Comment:
Do you want to include `FilterFn` and `AssertFn`?

Reply:
Hm yes I'll add some more headings for these.